dbt_semantic_interfaces/validations/primary_entity.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from typing import Generic, List, Sequence
+
+from dbt_semantic_interfaces.implementations.semantic_manifest import (
+    PydanticSemanticManifest,
+)
+from dbt_semantic_interfaces.protocols import SemanticManifestT, SemanticModel
+from dbt_semantic_interfaces.references import SemanticModelReference
+from dbt_semantic_interfaces.type_enums import EntityType
+from dbt_semantic_interfaces.validations.validator_helpers import (
+    FileContext,
+    SemanticManifestValidationRule,
+    SemanticModelContext,
+    ValidationError,
+    ValidationIssue,
+    validate_safely,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PrimaryEntityRule(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """Checks that the primary entity has been properly defined in a semantic model.
+
+    * If a semantic model contains dimensions, the primary entity must be available.
+    * The primary entity could be defined by the primary_entity field, or by one of the entities defined in a semantic
+      model.
+    * There should only be one primary entity in the model.
+    """
+
+    @staticmethod
+    def _model_requires_primary_entity(semantic_model: SemanticModel) -> bool:
+        return len(semantic_model.dimensions) > 0
+
+    @staticmethod
+    @validate_safely("Check that a semantic model has properly configured primary entities.")
+    def _check_model(semantic_model: SemanticModel) -> Sequence[ValidationIssue]:
+        context = SemanticModelContext(
+            file_context=FileContext.from_metadata(metadata=semantic_model.metadata),
+            semantic_model=SemanticModelReference(semantic_model_name=semantic_model.name),
+        )
+
+        # If there are entities defined in the model, check that there's only one primary entity.
+        entities_with_primary_time = tuple(
```

Comment:
nit: I'm confused why the word `time` is part of the variable name, is it meant to be `type`?

Reply:
Yeah, should be. Updated.